shared/utils/checkStatus.ts

```diff
@@ -0,0 +1,34 @@
+import { capitalize } from "./capitalize";
+import { Types } from "@requestnetwork/request-client.js";
+
+export const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+  const balance = Number(request?.balance?.balance ?? 0);
+  const expectedAmount = Number(request?.expectedAmount ?? 0);
+  const today = new Date();
+  const dueDate = new Date(request?.contentData?.paymentTerms?.dueDate);
+
+  const eventStatus = {
+    reject: "Rejected",
+    cancel: "Canceled",
+  };
+
+  if (balance > 0) {
+    return balance >= expectedAmount ? "Paid" : "Partially Paid";
+  }
+
+  if (dueDate < today && balance === 0) {
+    return "Overdue";
+  }
```

Comment:
Please change the logic so that Overdue status trumps Partially Paid.

Something like the following pseudocode:

```suggestion
  if (balance >= expectedAmount) {
    return "Paid";
  }
  else if (overdue && !pending && !canceled && !rejected) { 
    return "Overdue";
  }
  else if (balance > 0) {
    return "Partially Paid";
  }
  else if (canceled) {
    return "Canceled";
  }
  else if (rejected) {
    return "Rejected"
  }
  else if (request.state === "created") {
    return "Awaiting Payment";
  }
  else {
    // "Pending" or "Accepted"
    return capitalize(request.state)
  }
```

Reply:
@MantisClone We still need to go trough the Events objects and check the key name, we will just add more code, I added a simpler solution